src/components/legend.ts

```diff
@@ -326,6 +329,38 @@ export module Components {
       }
     }
 
+    /**
+     * Gets the function determining the opacity of the symbols of the Legend.
+     *
+     * @returns {(datum: any, index: number) => number}
+     */
+    public symbolOpacity(): (datum: any, index: number) => number;
```

Comment:
Upcasting makes the most sense -- the typing information supplied by Typescript should protect users in that language.